### PR TITLE
Add another potential fix for icon misalignment in doom-modeline

### PR DESCRIPTION
**Pull request by skyler544**
I've added the code I used to finally fix modeline icon misalignment
problems in my own setup as a suggestion in the Troubleshooting section.

### CODE REVIEW

File: modules/ui/modeline/README.org [hunk 1]
```diff
@@ -89,6 +89,33 @@ I believe the consensus is: this is due to oversized icons, i.e. a font issue. S
 
 (Mentioned in #1680, #278 and seagle0128/doom-modeline#334)
 
+4. If the previous suggestions don't fix the issue, try the following code:
```
Comment (hlissner) [inline modules/ui/modeline/README.org:92]: This summary is unhelpful. Change to

> 4. Change the width of icon characters in `char-width-table`:

**Review by hlissner — CHANGES_REQUESTED** [review]

File: modules/ui/modeline/README.org [hunk 2]
```diff
@@ -89,6 +89,33 @@ I believe the consensus is: this is due to oversized icons, i.e. a font issue. S
...
+           (set-char-table-range table char width))
+         (optimize-char-table table)
+         (set-char-table-parent table char-width-table)
+         (setq char-width-table table))))
+
+     (defun set-modeline-icon-width ()
+       (doom-modeline--set-char-widths doom-modeline-rhs-icons-alist))
+
+     (after! doom-modeline
+       (add-hook 'doom-modeline-mode-hook #'set-modeline-icon-width))
```
[the remark this answers is not in the document]
Comment (skyler544) [reply modules/ui/modeline/README.org]: As usual, your code is clearer and conciser than anything I can write. I've added a new commit with the requested changes.

File: modules/ui/modeline/README.org (hunk 1 above)
Comment (skyler544) [reply modules/ui/modeline/README.org]: Agreed, that's better. I've added this in a second commit.

**Review by skyler544 — COMMENTED** [review]
This commit contains the requested changes.

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Good work. Thanks for the help!

**Pull request merged**